cohort-engine/src/main/java/com/ibm/cohort/engine/cos/CosDao.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * (C) Copyright IBM Corp. 2021, 2021
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.cohort.engine.cos;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.ibm.cloud.objectstorage.services.s3.AmazonS3;
+import com.ibm.cloud.objectstorage.services.s3.model.Bucket;
+import com.ibm.cloud.objectstorage.services.s3.model.GetObjectRequest;
+import com.ibm.cloud.objectstorage.services.s3.model.S3Object;
+
+public class CosDao {
```

Comment:
This class might be a little "cart before the horse", but I'm OK keeping it around (but in a different project) to see if we get value out of it.

Reply:
Correct. The intent was to have this code be the backing implementation to a #future class called `CosCacheProvider`.